app.py

```diff
@@ -1,72 +1,179 @@
+    app.py reallocate_person <person_identifier> <room_name>
+    app.py load_people
+    app.py save_state [(--db sqlite_database)]
+    app.py load_state <sqlite_database>
+    app.py (-i | --interactive)
+    app.py (-h | --help | --version)
+
 Options:
     -i, --interactive  Interactive Mode
     -h, --help  Show this screen and exit.
 """
 
-#import necessary modules
 import sys
 import cmd
 from docopt import docopt, DocoptExit
 
+from dojo import Dojo
+
+
+def docopt_cmd(func):
+    """
+    This decorator is used to simplify the try/except block and pass the result
+    of the docopt parsing to the called action.
+    """
+
+    def fn(self, arg):
+        try:
+            opt = docopt(fn.__doc__, arg)
+
+        except DocoptExit as e:
+            # The DocoptExit is thrown when the args do not match.
+            # We print a message to the user and the usage block.
+
+            print('Invalid Command!')
+            print(e)
+            return
+
+        except SystemExit:
+            # The SystemExit exception prints the usage for --help
+            # We do not need to do the print here.
+
+            return
+
+        return func(self, opt)
 
+    fn.__name__ = func.__name__
+    fn.__doc__ = func.__doc__
+    fn.__dict__.update(func.__dict__)
+    return fn
 
 
-#for interactive shell define the class
-class MyInteractive (cmd.Cmd):
+class MyInteractive(cmd.Cmd):
     intro = 'Welcome to my interactive program!' \
         + ' (type help for a list of commands.)'
     prompt = 'dojo>>>'
     file = None
 
-    #use decorators to define the functions
     @docopt_cmd
     def do_create_room(self, arg):
-        """Usage: create_room <room_type> <room_names>..."""
-        try:
-            Dojo().create_room(arg.get('<room_type>'), arg.get('<room_names>'))
-            print ('end ...')
-        except TypeError:
-            print ('Not a valid room type try "living or office".')
-        except:
-            print('Not a valid usage')
+        """Usage: create_room <room_type> <room_names>...
+        """
+        room_purpose = arg.get('<room_type>')
+        room_names = arg.get('<room_names>')
+        Dojo().create_room(room_purpose, room_names)
 
     @docopt_cmd
     def do_add_person(self, arg):
-        """Usage: add_person (<fname> <lname> <FELLOW/STAFF> [<wants_accommodation>])
+        """Usage: add_person <fname> <lname> <FELLOW/STAFF> [<wants_accommodation>]
         """
-        try:
-            if(arg.get('<wants_accommodation>')):
-                Dojo().add_person(arg.get('<fname>'), arg.get('<lname>'), arg.get('<FELLOW/STAFF>'), arg.get('<wants_accommodation>','n')  )
-                print('end valid wants_accommodation...')
+        fname = arg.get('<fname>')
+        lname = arg.get('<lname>')
+        person_type = arg.get('<FELLOW/STAFF>')
+        wants_accommodation = arg.get('<wants_accommodation>')
+        if fname.isalpha() and lname.isalpha():
+            if person_type == 'staff':
```

Comment:
Might be a good idea to handle both upper and lower cases here.

The usage provided reads
```
app.py add_person <fname> <lname> <FELLOW/STAFF> [<wants_accommodation>]
```

Emphasis on **FELLOW/STAFF** (uppercased) in the usage but you are checking for the lowercased version in your code